### PR TITLE
24067: Fixed so we are now able to edit root units

### PR DESCRIPTION
**Pull request by cmoesgaard**
https://redmine.magenta-aps.dk/issues/24067

- [x] Have you updated the documentation?
- [x] Have you performed a smoke test of the application?
- [x] Have you written tests for your changes?
- [x] Have you updated the release notes?

### CODE REVIEW

File: NEWS.rst [hunk 1]
```diff
@@ -6,6 +6,11 @@ New features
 
 * #22849: Implemented support for signed SAML AuthN Requests.
 
+Bug fixes
+---------
+
+* #24067: Fixed being able to edit root organisational units
```
Comment (danchr) [inline NEWS.rst]: “Allowed editing root organisational units.”?

Strictly speaking, I wouldn't say this was a bug, but rather a missing feature.

File: backend/mora/service/orgunit.py [hunk 2]
```diff
@@ -106,6 +106,10 @@ def edit(self, unitid, req):
             original, mapping.PARENT_FIELD.path)[-1]
         parent_uuid = util.get_uuid(parent)
 
+        org = util.get_obj_value(
```
Comment (danchr) [inline backend/mora/service/orgunit.py]: Don't we have a `get_obj_uuid` (or something) for exactly this case?

File: backend/mora/service/orgunit.py [hunk 3]
```diff
@@ -158,8 +162,9 @@ def edit(self, unitid, req):
         payload = common.ensure_bounds(new_from, new_to, bounds_fields,
                                        original, payload)
 
-        validator.is_date_range_in_org_unit_range(parent_uuid, new_from,
-                                                  new_to)
+        if org_uuid != parent_uuid:
+            validator.is_date_range_in_org_unit_range(parent_uuid, new_from,
+                                                      new_to)
```
Comment (danchr) [inline backend/mora/service/orgunit.py]: We should probably validate that we're not exceeding the organisation's validity, shouldn't we?

File: backend/mora/service/orgunit.py [hunk 4]
```diff
@@ -593,7 +598,7 @@ def edit_org_unit(unitid):
     to contain the fields that need to change along with the validity dates.
 
     :<json string name: The name of the org unit
-    :<json string parent: The parent org unit
+    :<json string parent: The parent org unit or organisation
```
Comment (danchr) [inline backend/mora/service/orgunit.py]: This seems to indicate that we can move units to/from the root level. Should that be the case? Do we test it?

File: NEWS.rst (hunk 1 above)
Comment (cmoesgaard) [reply NEWS.rst]: We never mentioned _not_ being able to edit root organisational units, and the error message wasn't particularly helpful - so I think this qualifies as a bug.

File: backend/mora/service/orgunit.py (hunk 2 above)
Comment (cmoesgaard) [reply backend/mora/service/orgunit.py]: We have a ton of `get_*` functions, but not one for this particular purpose, AFAIK.

File: backend/mora/service/orgunit.py (hunk 3 above)
Comment (cmoesgaard) [reply backend/mora/service/orgunit.py]: I think organisations are always just created to last forever, if we're suddenly outside the validity period of the organisation, we have bigger problems.

File: backend/mora/service/orgunit.py (hunk 4 above)
Comment (cmoesgaard) [reply backend/mora/service/orgunit.py]: Yes, I don't see why not. And no, but I'll add some.

File: backend/mora/service/orgunit.py (hunk 3 above)
Comment (danchr) [reply backend/mora/service/orgunit.py]: Well, that depends on the importer — for example, most municipalities have a from day one might use. A simple `TODO` might suffice?

File: backend/mora/service/orgunit.py (hunk 4 above)
Comment (danchr) [reply backend/mora/service/orgunit.py]: My main worry is that whatever we do, we:

1) Know we're doing it.
2) Have tests for it.
3) On some level, it makes sense.

I'm willing to compromise on the last item as long as the two previous are covered 😁  But yes, moving from the root level makes sense, but is somewhat hard to test as it presumes a different hierarchy. Given that the easiest way to test it is moving a unit back and forth, we might as well do both IMO.

Comment (cmoesgaard) [reply backend/mora/service/orgunit.py]: So you're suggensting we remove the existing validiation rule preventing us from moving from the root level?

Comment (danchr) [reply backend/mora/service/orgunit.py]: I think we should ask @alexthirifays — they change seems fairly simple, but I can't quite settle on whether the users would want it. They probably would.

Comment (danchr) [reply backend/mora/service/orgunit.py]: And the conclusion from asking him is that it seems a rather risky operation, so out of an abundance of caution we should disallow it, for now. In other words: Re-parenting a unit to or from the root is forbidden.

Comment (cmoesgaard) [reply backend/mora/service/orgunit.py]: Alright. I'll add an additional validation to ensure we aren't able to move _to_ the root level as well.

**Comment by danchr** [comment]
🏇

**Pull request merged**